### PR TITLE
Fix :not CSS rule syntax in post detail webview

### LINKED ISSUE #4302: p, div, li line-height broken for reader post detail on tablets
The `<p>`, `<div>` and `<li>` elements should have line-height = 1.6em on tablets but that rule is broken because of #4222 

The lines broken are: https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostRenderer.java#L328
and
https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostRenderer.java#L335

### PR DESCRIPTION
**Pull request by hypest**
Fixes #4302

Unfortunately, the :not selector does not support cascading and it's unsuitable for excluding all tiled-gallery classes. Instead, this fix attaches a new class to the gallery elements and use that to uniquely exclude them.

To test:
* Open a tiled-gallery post in the Reader
* Notice that post body text is now styled correctly (line-height and font-size)

Needs review: @nbradbury

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostRenderer.java [hunk 1]
```diff
@@ -307,6 +310,9 @@ private void replaceIframeTag(final String tag, final String src) {
      * returns the full content, including CSS, that will be shown in the WebView for this post
      */
     private String formatPostContentForWebView(final String content, boolean renderAsTiledGallery) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostRenderer.java]: The changes below look good, but I'm wondering if we can also change the `renderAsTiledGallery` parameter so it's only true when the post actually contains a gallery.

Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostRenderer.java]: Cool, what do you have in mind Nick? Something like searching for `tiled-gallery` in the content html? It has to be something simple and accurate.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostRenderer.java]: It looks like the most reliable technique is to check whether the `format` field of the reader post is "gallery."

Unfortunately, we haven't been storing `format` - but as luck would have it, I added it just today in PR #4316 along with an <a href="https://github.com/wordpress-mobile/WordPress-Android/pull/4316/files#diff-0bd1e069dded619660267aa1a74332cfR367">isGallery()</a> method to check whether a post contains a gallery.

So once that PR is merged, you can use that method.

Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostRenderer.java]: Aha, thanks for the info @nbradbury .

I can be wrong but it seems that the post format can be user configurable and I wouldn't exactly feel "safe" to use it if so.

The fix here (and the parent PR) is directly related to tiled-galleries and is only in effect when certain CSS classes exist so I think the easiest discriminator is the CSS class name itself. I will add a quick lookup for the primary class name (`.tiled-gallery`) and we can change later if the flag you mention proves to be a better choice. Thanks!

**Comment by hypest** [comment]
Ready for another pass Nick, thanks!

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**